File: chat-frontend/lib/brand.ts
// Brand names, in one place. Every user-visible occurrence of the product or
// company name should import from here rather than hard-coding a string, so a
// rename is an edit to this file rather than a sweep across the app.
//
// The backend has a matching module at backend/core/brand.py — keep them in
// step. (They are separate files rather than one served value because these
// strings appear in server-rendered <title> metadata, before any fetch.)

// The product.
export const BRAND_NAME = "Hudson Corpus";

// The company — used where we speak as the operator (Terms, the footer).
export const COMPANY_NAME = "Hudson Legal Technologies";

// The MCP server's connector key — the name users type into `claude mcp add`
// and that lands in their local client config. Renaming it breaks every existing
// connector, which is why it is being renamed NOW, while there are zero
// registered clients, and frozen permanently after: it must survive any future
// rebrand, so treat it as a wire identifier and not as a brand string.
export const MCP_SERVER_ID = "hudson-corpus";

// --- Origins ---------------------------------------------------------------
// This app is served FROM the app origin, so it can derive its own URL and needs
// no hard-coded domain. NEXT_PUBLIC_APP_URL (inlined at build time) wins when set
// — it is the only source available during SSR, where `window` does not exist.
// These are functions, not module consts, precisely because of that: the module
// is imported by server-rendered metadata, and a const would freeze in the
// window-less value.

/** Absolute origin the app is served from ("https://app.hudsonlegal.tech"), or "". */
export function appOrigin(): string {
  const configured = process.env.NEXT_PUBLIC_APP_URL;
  if (configured) return configured.replace(/\/+$/, "");
  if (typeof window !== "undefined") return window.location.origin;
  return "";
}

/** Host portion of the app origin ("app.hudsonlegal.tech"), for display. */
export function appHost(): string {
  const origin = appOrigin();
  if (!origin) return "";
  try {
    return new URL(origin).host;
  } catch {
    return origin.replace(/^https?:\/\//, "");
  }
}

/** The MCP endpoint external clients connect to, on this same origin. */
export function mcpUrl(): string {
  return `${appOrigin()}/mcp`;
}

File: backend/core/brand.py
"""Brand names, in one place.

Every user-visible occurrence of the product or company name should import from
here rather than hard-coding a string. Renaming the product then means editing
this file, not sweeping ~90 strings across two frontends and a Django app.

Note what is deliberately NOT here: the Django app label ``apps.corpus`` and its
tables. "Corpus" there means *the body of law* — a domain term that survives any
rebrand — not the brand.
"""

from __future__ import annotations

# The product: what a user sees in the app, on the consent screen, in emails.
BRAND_NAME = "Hudson Corpus"

# The company. Legal entity name; used where we speak as the operator (Terms,
# contracts, the "operated by" line).
COMPANY_NAME = "Hudson Legal Technologies"

# The MCP server's wire identifier, advertised as serverInfo.name and used as the
# connector key in the install snippet.
#
# Renamed from "iowa-legal-corpus" on 2026-07-13, during the zero-client window:
# no registered OAuth clients and no live tokens existed, so no local client
# config anywhere held the old string and nothing could be orphaned.
#
# That window is now closed. From here this value is FROZEN PERMANENTLY: the
# moment a real client writes it into a config file on a machine we cannot reach
# (`claude mcp add ... hudson-corpus`, claude_desktop_config.json), changing it
# silently breaks their connector with no server-side fix. It must not follow a
# future BRAND_NAME change — it is an opaque key, not a name.
MCP_SERVER_ID = "hudson-corpus"

# How we identify ourselves to servers we crawl (polite-crawler contact string).
CRAWLER_CONTACT = "nick@nickhudson.me"
